app/src/main/java/com/openvehicles/OVMS/ui2/pages/HomeFragment.kt

```diff
@@ -1036,7 +1041,28 @@ class HomeFragment : BaseFragment(), OnResultCommandListener, HomeTabsAdapter.It
     private fun initialiseTabs(carData: CarData?) {
         tabsAdapter.mData = emptyList()
 
-        tabsAdapter.mData += HomeTab(TAB_CONTROLS, R.drawable.ic_controls_tab, getString(R.string.controls_tab_label), null)
+        var tpms = ""
+        if(appPrefs.getData("showtpmscontrol", "off") == "on"){
+            var pressure = arrayOf(
+                carData?.car_tpms_fl_p,
+                carData?.car_tpms_fr_p,
+                carData?.car_tpms_rl_p,
+                carData?.car_tpms_rr_p
+            )
+            tpms = String.format(
+                "%s %s | %s %s | %s %s | %s %s",
+                getString(R.string.fl_tpms),
+                pressure?.get(0) ?: "",
+                getString(R.string.fr_tpms),
+                pressure?.get(1) ?: "",
+                getString(R.string.rl_tpms),
+                pressure?.get(2) ?: "",
+                getString(R.string.rr_tpms),
+                pressure?.get(3) ?: ""
+            )
```

Comment:
This whole section relies on the deprecated "W" TPMS message, which cannot adapt to other wheel layouts and will be removed in the future. Please use the flexible new "Y" message arrays, and take into account that not all vehicles have four wheels. Test your code e.g. with an Energica bike or manually crafted wheel layouts.